cmd/snap-bootstrap/partition/sfdisk.go

```diff
@@ -103,14 +103,24 @@ func (dl *DeviceLayout) CreateMissing(pv *gadget.LaidOutVolume) ([]DeviceStructu
 
 	// Write the partition table, note that sfdisk will re-read the
 	// partition table by itself: see disk-utils/sfdisk.c:write_changes()
-	cmd := exec.Command("sfdisk", dl.Device)
+	cmd := exec.Command("sfdisk", "--force", dl.Device)
```

Comment:
I've looked at sfdisk source code, the flag allows the changes to proceed even if the underlying disk device is in use. I understand that in the actual usage scenario, snap-bootstrap, will be running off of an initramfs, while the disk being modified will not be used. 

If the change is just for a spread test, then I think it's better to hide it behind an environment variable, eg. SNAPD_BOOSTRAP_FORCE or something similar.

Reply:
The `--force` flag is needed in the production case. At that point the system is seeded and we have existing partitions mounted at that point (they were mounted in initramfs, partitioning happens post-initramfs).

Reply:
Reading the man page it has somewhat concerning description:
```
       -f, --force
              Disable all consistency checking.
```

Let me read what this code path is doing.

Reply:
@bboozzoo i don't quote follow your comments here.

the gadget.yaml is on the partition of the disk to be partitioned. So by definition, the disk will be in use whenever we are partitioning it. Unless one mounts the seed partition, copies gadget yaml somewhere else, and then unmounts the seed partition.

IMHO initrd shouldn't include recovery/install code, as all of that should be done out of the booted recovery. Meaning initrd will only have snap-bootstrap's initramfs-mounts but not the partitioning commands. Otherwise I will have to carry partitioning tools in all the initrds, when most of the time it is not used. But this is to be determined.